src/unstract/sdk/utils/retry_utils.py

```
@@ -0,0 +1,306 @@
+                    if retry_predicate is not None:
+                        should_retry = retry_predicate(e)
+
+                    # Check if we've exceeded max time
+                    elapsed_time = time.time() - start_time
+                    if elapsed_time >= max_time:
+                        logger_instance.error(
+                            "Giving up '%s' after %.1fs (max time exceeded): %s",
+                            func.__name__,
+                            elapsed_time,
+                            e,
+                        )
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)